calc_test.py

```diff
@@ -1,52 +1,56 @@
-# TODO: finish adding tests for all major functions in calc.py
-# test suite below
-def test_add():
-    assert add(1, 2) == 3
-    assert add(0, 0) == 0
-    assert add(-1, 1) == 0
-    assert add(1, -1) == 0
-    assert add(-1, -1) == -2
-
-def test_subtract():
-    assert subtract(5, 3) == 2
-    assert subtract(5, -3) == 8
-    assert subtract(-4, 3) == -7
-    assert subtract(-4, -3) == -1
-    assert subtract(0, 0) == 0
-    assert subtract(0, -1) == 1
-
-def test_multiply():
-    assert multiply(5, 3) == 15
-    assert multiply(5, -3) == -15
-    assert multiply(-4, 0) == 0
-    assert multiply(3, 0) == 0
-    assert multiply(0, 0) == 0
-    assert multiply(-1, -5) == 5
-
-def test_divide():
-    assert divide(10, 2) == 5
-    assert divide(10, -2) == -5
-    assert divide(-10, 2) == -5
-    assert divide(-5, 5) == -1
-    assert divide(0, 1) == 0
-
-def test_exponent():
-    assert exponent(2, 3) == 8
-    assert exponent(2, -2) == 0.25
-    assert exponent(2, 0) == 1
-    assert exponent(0, 0) == 1
-    assert exponent(-2, 3) == -8
-
-def test_modulo():
-    assert modulo(10, 3) == 1
-    assert modulo(10, 4) == 2
-    assert modulo(10, 1) == 0
-
-def test_sqrroot():
-    assert sqrroot(9) == 3
-    assert sqrroot(25) == 5
-    assert sqrroot(0) == 0
-    assert sqrroot(1) == 1
\ No newline at end of file
+import unittest
+from calc import *
+
+class calc_test(unittest.TestCase):
+    # TODO: finish adding tests for all major functions in calc.py
+    # test suite below
+    def test_add(self):
+        assert add(1, 2) == 3
+        assert add(0, 0) == 0
+        assert add(-1, 1) == 0
+        assert add(1, -1) == 0
+        assert add(-1, -1) == -2
+    
+    def test_subtract(self):
+        assert subtract(5, 3) == 2
+        assert subtract(5, -3) == 8
+        assert subtract(-4, 3) == -7
+        assert subtract(-4, -3) == -1
+        assert subtract(0, 0) == 0
+        assert subtract(0, -1) == 1
+    
+    def test_multiply(self):
+        assert multiply(5, 3) == 15
+        assert multiply(5, -3) == -15
+        assert multiply(-4, 0) == 0
+        assert multiply(3, 0) == 0
+        assert multiply(0, 0) == 0
+        assert multiply(-1, -5) == 5
+    
+    def test_divide(self):
+        assert divide(10, 2) == 5
+        assert divide(10, -2) == -5
+        assert divide(-10, 2) == -5
+        assert divide(-5, 5) == -1
+        assert divide(0, 1) == 0
+    
+    def test_exponent(self):
+        assert exponent(2, 3) == 8
+        assert exponent(2, -2) == 0.25
+        assert exponent(2, 0) == 1
+        assert exponent(0, 0) == 1
+        assert exponent(-2, 3) == -8
+    
+    def test_modulo(self):
+        assert modulo(10, 3) == 1
+        assert modulo(10, 4) == 2
+        assert modulo(10, 1) == 0
+    
+    def test_sqrroot(self):
+        assert sqrroot(9) == 3
+        assert sqrroot(25) == 5
+        assert sqrroot(0) == 0
+        assert sqrroot(1) == 1
+
+if __name__ == '__main__':
+    unittest.main()

```

Comment:
Just wondering, what would this change do? Pytest works just fine on what I wrote by myself, would it still work with your changes?

Reply:
Oh, this is for unittest, which is the other frame work for testing in python.

Reply:
I would definitely prefer pytest, it seems more straightforward.